gtk/gtk.go

```diff
@@ -9506,6 +9516,254 @@ func (v *TreeRowReference) Valid() bool {
 	return gobool(c)
 }
 
+/*
+ * GtkTreeSortable
+ */
+
+// TreeSortable is a representation of GTK's GtkTreeSortable
+type TreeSortable struct {
+	*glib.Object
+}
+
+// ITreeSortable is an interface type implemented by all structs
+// embedding a TreeSortable.  It is meant to be used as an argument type
+// for wrapper functions that wrap around a C GTK function taking a
+// GtkTreeSortable.
+type ITreeSortable interface {
+	toTreeSortable() *C.GtkTreeSortable
+}
+
+// native returns a pointer to the underlying GtkTreeSortable
+func (v *TreeSortable) native() *C.GtkTreeSortable {
+	if v == nil || v.GObject == nil {
+		return nil
+	}
+	p := unsafe.Pointer(v.GObject)
+	return C.toGtkTreeSortable(p)
+}
+
+func (v *TreeSortable) toTreeSortable() *C.GtkTreeSortable {
+	if v == nil {
+		return nil
+	}
+	return v.native()
+}
+
+func marshalTreeSortable(p uintptr) (interface{}, error) {
+	c := C.g_value_get_object((*C.GValue)(unsafe.Pointer(p)))
+	obj := glib.Take(unsafe.Pointer(c))
+	return wrapTreeSortable(obj), nil
+}
+
+func wrapTreeSortable(obj *glib.Object) *TreeSortable {
+	return &TreeSortable{obj}
+}
+
+// TreeIterCompareFunc defines the function prototype for the sort function (f arg) for
+// (* TreeSortable).SetSortFunc
+type TreeIterCompareFunc func(model *TreeModel, a, b *TreeIter, userData interface{}) int
+
+// GetSortColumnId() is a wrapper around gtk_tree_sortable_get_sort_column_id().
+func (v *TreeSortable) GetSortColumnId() (int, SortType, bool) {
+	sort := C.toGtkTreeSortable(unsafe.Pointer(v.native()))
+	var column C.gint
+	var order C.GtkSortType
+	ok := gobool(C.gtk_tree_sortable_get_sort_column_id(sort, &column, &order))
+	return int(column), SortType(order), ok
+}
+
+type treeStoreSortFuncData struct {
+	fn       TreeIterCompareFunc
+	userData interface{}
+}
+
+var (
+	treeStoreSortFuncRegistry = struct {
+		sync.RWMutex
+		next int
+		m    map[int]treeStoreSortFuncData
+	}{
+		next: 1,
+		m:    make(map[int]treeStoreSortFuncData),
+	}
+)
+
+// SetSortColumnId() is a wrapper around gtk_tree_sortable_set_sort_column_id().
+func (v *TreeSortable) SetSortColumnId(column int, order SortType) {
+	sort := C.toGtkTreeSortable(unsafe.Pointer(v.native()))
+	C.gtk_tree_sortable_set_sort_column_id(sort, C.gint(column), C.GtkSortType(order))
+}
+
+// SetSortFunc() is a wrapper around gtk_tree_sortable_set_sort_func().
+func (v *TreeSortable) SetSortFunc(sortColumn int, f TreeIterCompareFunc, userData ...interface{}) error {
+	if len(userData) > 1 {
+		return errors.New("userData len must be 0 or 1")
+	}
+
+	t := treeStoreSortFuncData{fn: f}
+	if len(userData) > 0 {
+		t.userData = userData[0]
+	}
```

Comment:
same as below:

you can implement userData as type uintptr -> then you won't need a return value

see my pull request #511